crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_spread.rs

```diff
@@ -77,22 +77,54 @@ fn unnecessary_spread_fix(
     if let Some(last) = dict.iter_values().last() {
         // Ex) `**{a: 1, b: 2}`
         let mut edits = vec![];
+        let mut open_parens: u32 = 0;
+
+        for tok in SimpleTokenizer::starts_at(doublestar.end(), locator.contents()).skip_trivia() {
```

Comment:
I haven't looked at the complete solution but I'm wondering if `SimpleTokenizer` is required when we have access to the parsed tokens. You can access tokens for a [specific range](https://github.com/astral-sh/ruff/blob/6e9ff445fd8559972b423370de20563a9c2db8d4/crates/ruff_python_parser/src/lib.rs#L419-L419) or [all tokens after a specific offset](https://github.com/astral-sh/ruff/blob/6e9ff445fd8559972b423370de20563a9c2db8d4/crates/ruff_python_parser/src/lib.rs#L457-L457). Refer to the references of those methods as an example on how to use them.

Reply:
Oh, this rule already utilized `SimpleTokenizer` from before. Hmm, I wonder if we can remove that and switch to just getting the lexed tokens.